src/init/mod.rs

```diff
@@ -17,24 +17,24 @@ using whatever mechanism is available in the host shell--this two-phase solution
 has been developed as a compatibility measure with `eval $(starship init X)`
 */
 
-fn path_to_starship() -> io::Result<String> {
-    let current_exe = env::current_exe()?
-        .to_str()
-        .ok_or_else(|| io::Error::new(io::ErrorKind::Other, "can't convert to str"))?
-        .to_string();
-    Ok(current_exe)
+fn path_to_starship() -> String {
+    env::args()
+        .next()
+        .unwrap_or_else(|| String::from(crate_name!()))
```

Comment:
I'm curious, for the fallback why use `crate_name` rather than the current `env::current_exe` approach?

Reply:
TBH it shouldn't ever happen unless someone explicitly messes with `argv0` like using `exec -a something starship...`. I put it there just for the sake of having avoiding forceful unwrapping.
Now when you ask me it might be better to use `env::current_exe` but it has both pros and cons compared to `crate_name`. `crate_name` is resolved during compile time which is obviously plus and it is deterministic regardless of platform. Yes, it might be the wrong thing to fall back onto because starship might not be available on the path. OTOH, `env::current_exe` is runtime call and completely unreliable (https://doc.rust-lang.org/std/env/fn.current_exe.html#platform-specific-behavior). So the real question is should we fall back to:
1. something that might be wrong but at least is always the same
2. something that might be wrong but is pretty much a moving target
3. should we ditch this altogether and proclaim it will never happen 🤔